core/embed/boardloader/main.c

```diff
@@ -68,6 +68,13 @@ static const uint8_t * const BOARDLOADER_KEYS[] = {
 
 #ifdef STM32U5
 void check_bootloader_version(uint8_t bld_version) {
+  if (sectrue != secret_verify_header()) {
```

Comment:
It would be better to move this code to a dedicated function. Clearing invalid secrets does not relate to bootloader version checking.

Reply:
moved https://github.com/trezor/trezor-firmware/commit/0153fba65a000ac74667dae0b7e52551bfc66d6e